deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/base/MatchIndex.java

```diff
@@ -377,4 +393,33 @@ public int getMatchIndex() {
     }
 
   }
+
+  /**
+   * Encapsulates the result of the compare operation.
+   */
+  private static class CompareResult {
```

Comment:
I did not make it a sealed class but moved it to Kotlin. I kept it in Java because of it being a static inner class is is a very lean construct. I think the same is true for the Kotlin data class.